src/rust/ide/src/controller/graph/executed.rs

```diff
@@ -37,15 +59,36 @@ pub enum Notification {
 /// Handle providing executed graph controller interface.
 #[derive(Clone,CloneRef,Debug)]
 pub struct Handle {
+    #[allow(missing_docs)]
+    pub logger:Logger,
     /// A handle to basic graph operations.
-    pub graph:controller::Graph,
+    graph:Rc<RefCell<controller::Graph>>,
     /// Execution Context handle, its call stack top contains `graph`'s definition.
     execution_ctx:Rc<ExecutionContext>,
+    /// The handle to project controller is necessary, as entering nodes might need to switch
+    /// modules, and only the project can provide their controllers.
+    project:controller::Project,
+    /// The publisher allowing sending notification to subscribed entities. Note that its outputs is
+    /// merged with publishers from the stored graph and execution controllers.
+    notifier:Rc<RefCell<crate::notification::Publisher<Notification>>>,
```

Comment:
One more thought: perhaps we don't need a Rc for publisher, as it implements CloneRef. Moreover, perhaps we can make our wrapper in notification::Publisher actually implement the internal mutability pattern, so we could also resign from RefCell here as well (and possibly from other places, like model::Module).

Reply:
> as it implements CloneRef
I don't think it does? 

But likely it should.

Reply:
refactored